src/index.ts

```diff
@@ -26,25 +40,88 @@ const defaultOptions: MTROptions = {
  */
 export default async function watchAndNotify(mongoData: MongoDataType, rabbitData: RabbitDataType, opts?: Partial<MTROptions>): Promise<void> {
     const options: MTROptions = { ...defaultOptions, ...opts };
-    if (opts?.middleware && !options.prettify) {
-        console.log('MTR: ===> error: middleware option cannot work when prettify is false');
-        return;
+    rabbitData.queues.forEach(queue => {
+        if (!options.prettify && queue.middleware !== undefined) {
+            console.log('MTR: ===> error: middleware option cannot work when prettify is false');
+            return;
+        }
+    })
+
+    await initRabbitConn(rabbitData.rabbitURI, options);
+    await initQueues(getQueuesNames(rabbitData.queues));
+
+    log(`successful connection to all queues`, options);
+    log(`connecting to mongo collection: ${mongoData.collectionName} with connectionString ${mongoData.connectionString} ...`, options);
+    
+    mongoConn = new MongoClient(mongoData.connectionString, {useUnifiedTopology: true});
+    mongoConnection(mongoData, rabbitData, options);
+}
+
+/**
+ * Get rabbit health status
+ * @returns boolean - true if rabbit is healthy
+ */
+export function getRabbitHealthStatus(): boolean{
+    return !menash.isClosed && menash.isReady;
+}
+
+/**
+ * Get mongo connection health status
+ * @returns boolean - true if mongo is healthy
+ */
+export function getMongoHealthStatus(): boolean {
+    return mongoConn.isConnected();
+}
+
+/**
+ * Creates mongodb connection.
+ * @param mongoData  - mongo data - collections and connection uri (MongoDataType)
+ * @param rabbitData - rabbit data (RabbitDataType)
+ * @param options    - contains the MTROptions.
+ */
+async function mongoConnection(mongoData: MongoDataType, rabbitData: RabbitDataType, options: MTROptions) {
+    while(true) {
+        try {
+            log('try connect to mongo', options);
+            await mongoConn.connect();
+            log(`successful connection to mongo on connectionString: ${mongoData.connectionString}`, options);
```

Comment:
maybe make these console.log, so it will always print upon connection / error?

Reply:
done